### PR TITLE
Remove unneeded patch to dbconsole

### PR DESCRIPTION
**Pull request by paarthmadan**
Dependent on https://github.com/rails/rails/pull/46093

With https://github.com/rails/rails/pull/46093, the dbconsole implementation will consult the adapter to initialize a console to the database. Because `TrilogyAdapter` inherits from `AbstractMysqlAdapter`, this logic is already defined and thus requires no further changes outside of removing the original patch.

I'd assume the dbconsole changes make their way into Rails 7.2, so this change should be gated behind the next release where the expectation on `active_record`'s version is bumped to 7.2 as well.

cc: @gmcgibbon

### CODE REVIEW

File: lib/trilogy_adapter/connection.rb [hunk 1]
```diff
@@ -14,6 +14,10 @@ module TrilogyAdapter
   #                                           host: "localhost",
   #                                           database: "demo_development"
   module Connection
+    def trilogy_connection_class
```
Comment (eileencodes) [inline lib/trilogy_adapter/connection.rb:17]:
```suggestion
    def trilogy_adapter_class
```

File: lib/trilogy_adapter/connection.rb [hunk 2]
```diff
@@ -31,7 +35,7 @@ def trilogy_connection(config)
         0
       ]
 
-      ActiveRecord::ConnectionAdapters::TrilogyAdapter.new nil, logger, options, configuration
+      trilogy_connection_class.new nil, logger, options, configuration
```
Comment (eileencodes) [inline lib/trilogy_adapter/connection.rb:38]:
```suggestion
      trilogy_adapter_class.new nil, logger, options, configuration
```

**Review by eileencodes — COMMENTED** [review]
After this was merged on Rails I realized that the method name needed to change. `connection_class` already has a meaning so I changed it to `adapter_class` in https://github.com/rails/rails/pull/46166

**Comment by paarthmadan** [comment]
@eileencodes Sounds good, thanks for following up on this PR! Updated this PR to conform to the new names.

**Review by HParker — APPROVED** [review]
Happy to see that dbconsole patch go!

**Pull request merged**